libs/common/irohad_version.hpp

```diff
@@ -6,11 +6,30 @@
 #ifndef LIBS_COMMON_IROHAD_VERSION_HPP
 #define LIBS_COMMON_IROHAD_VERSION_HPP
 
+// disabling GNU macros
+#ifdef major
+#undef major
+#endif
+
+#ifdef minor
+#undef minor
+#endif
+
 namespace iroha {
 
   /// A string describing current git repository version in a human-readable way
   extern const char *kGitPrettyVersion;
 
+  struct IrohadVersion {
+    unsigned int major;
+    unsigned int minor;
+    unsigned int patch;
+
```

Comment:
union IrohaVersion {
  struct {
    uint8_t major;
    uint8_t minor;
    uint8_t patch;
    uint8_t reserved; // or smth else or may be patch 16 bit
  };
  uint32_t quad;
};
?

Reply:
nice, but imho not needed here.